### ISSUE TITLE
Customization option for assigning colors to chart values

### ISSUE THREAD

**Issue by Stefan781** [apache/superset#25464]
I have a custom color palette that I would like to use to represent different groups in my charts. Here is the palette:
<img width="179" alt="image" src="https://github.com/apache/superset/assets/130654984/92a9496f-8d73-4dc7-89ea-7315ca77dc25">
I have values that are dynamically calculated. I would like to assign the first color to the first value, based on some predefined condition. For example, alphabetical order. I have tried achieving this in multiple ways. When it comes to DeckGL graphs, it proved impossible, since the colors are assigned to their corresponding value in some arbitrary way, which is, to my disappointment, not controlled by the order defined in my dataset, created with a custom SQL query. When it comes to a pie chart, you have the option to _sort by metric_. With some "hacking" of the metric in a way that assigns a number based on the group so that the groups can be sorted in order, I get the desired result:
![image](https://github.com/apache/superset/assets/130654984/7f89b69e-21e8-41c9-bb56-aaaf9931f09c)
However, if the metric changes to something meaningful, such as COUNT(*), the colors are still assigned based on the metric, not on my custom order:
![image](https://github.com/apache/superset/assets/130654984/a9420a13-8f8a-4a9a-a7a6-122d0d9c3218)
Furthermore, even with the first approach, when applying filters, the color order breaks again:
![image](https://github.com/apache/superset/assets/130654984/5a71525d-bc95-4cfa-ae0b-bb9837b2c874)

What I would like to have is a precise rule that defines how are the colors assigned. As a matter of fact, if the colors were assigned based on the defined ORDER BY in the SQL code, I would get my desired result. Are there any workarounds to this problem? Is my approach wrong and is there a better one? If not, are there plans to give more freedom when it comes to assigning colors to the charts?

There is also the option to "hardcode" the color from the Dashboard settings. This, unfortunately, will not work in my case, since the groups will change dynamically if, for example, the max value becomes 20.

I see different use cases for this functionality. The two cases I am interested in are:
1. Monochrome colors for points on a Scatterplot DeckGL visualization based on some metric (for example, older points get a pale color). I actually came up with a workaround to this use case using a rather involved custom JS Interceptor but it still doesn't solve the problem if the user wants a specific color.
2. Monochrome/Analogic colors in Part of a Whole charts for cases where the range of values has a different order than the metric that is used to order them and the represented values can change dynamically.

**Comment by Stefan781** [comment]
Might be related to [#18392](https://github.com/apache/superset/discussions/18392#discussion-3850181)

**Comment by rusackas** [comment]
Closing this since it's a question rather than a bug, but you can do this in the dashboard properties editor, assigning colors to series values. Hope that helps a little :)

**Comment by Stefan781** [comment]
Are you sure that this is possible at all? If you know that it can work for dynamic values, please provide documentation or an example. Otherwise, it seems to me you are just closing the issue to keep the count low. @rusackas

**Comment by Stefan781** [comment]
[Here](https://preset.io/blog/customizing-chart-colors-with-superset-and-preset/) is the particular "tutorial" that probably does what you refer to. Notice how the labels are predefined (static).

**Comment by rusackas** [comment]
You're right on several counts:
• I misinterpreted the dynamic nature of the issue. 
• That is indeed the article I wrote with the technique I was referring to. Sorry if it's not up to your standards.
• We are indeed trying to keep the issue count at a sustainable level of actionable tasks. We're finding that this is actually improving engagement. I'll happily reopen the issue, as I always do when asked. 

However, this is not a bug, but rather a feature request... and a good one at that. Therefore, I'll convert this to a GitHub Discussion, which is where these conversations should land. If you want to contribute to a proposal/implementation of this, that would be appreciated, but it _might_ warrant a SIP depending on the complexity. In my mind, it comes down to one question:

Is the main goal to have the pie chart display as a gradient? If so, it seems like making the Pie chart support a linear/gradient palette (by sort order) is a reasonable addition that shouldn't require a SIP. If, however, the proposal is to use categorical palettes in a more deterministic order, that goes counter to the "color consistency" efforts where these charts may be assigned according to dashboard use. That gets exponentially more complex to solve.
